modules/system/ServiceProvider.php

```diff
@@ -570,7 +570,7 @@ protected function registerAssetBundles()
         });
 
         MixAssets::registerCallback(function ($mix) {
-            $mix->registerPackage('snowboard', '~/modules/system/assets/js/snowboard');
+            $mix->registerPackage('snowboard', '~/modules/system/assets/js/snowboard/winter.mix.js');
```

Comment:
If we're going to be assuming a package has a `winter.mix.js` file, should we need to define that filename when registering the package? I think, even in the case of multiple packages under the one extension, we should be fine with assuming the end target is `winter.mix.js`, therefore they only need to specify the path to the directory where that file resides.

Reply:
Yeah I went back and forth on this but ultimately what's the point in allowing them to drop the file? It's a mere 13 characters difference one time to write it out for much greater clarity, especially when searching the project for uses of the file. Additionally this manual registration is only required when the developer is going outside the norm anyways because of auto registration.

Reply:
No, I played around with that but it was causing the code to be messier and more complex than I liked. Currently it will complain if a valid .js file isn't provided which I think is fine, if you're going with convention (auto registration) then it's easy for you, if you're breaking convention then you need to be explicit. Comes back to the "searching the project for uses of the file" piece, the convention is clearly documented and easy to understand, but supporting half-baked manual registrations is of no benefit to anyone IMO. It was a good solution for the initial approach, but now that we have autoregistration it doesn't really serve any purpose to support that IMO.

<img width="1294" alt="Screen Shot 2022-02-13 at 10 49 19 AM" src="https://user-images.githubusercontent.com/7253840/153765529-73b42d05-0146-47c2-93e4-c73468ba0e21.png">